### DISCUSSION TITLE
custom menu video & audio

### DISCUSSION THREAD

**Discussion by fahimbench** [sibiraj-s/ngx-editor#395] [Q&A] [answered]
Hi, first i would thanks for this editor that works directly when integrate. 

I was wondering about to integrate videos (from youtube or direct sources) and audios (from sources) for my project but i don't understand well how to do it despite the documentation about the custom menu. I was looking to do like image integrations with link or upload but i don't find anything online.

So i guess its not already integrate ? 

I tried with insertHTML command but that remove all tags and keeps the text only.

If someone can help me with some clues, thanks by advance

**Comment by fahimbench** [comment, marked as answer]
nevermind, i did it successfully 
![Capture d’écran 2022-05-02 164116](https://user-images.githubusercontent.com/38366191/166253803-9e0c686c-4469-4a77-9456-6024f418c996.png)

**Reply by sibiraj-s** [reply]
Awesome. Sorry couldn't reply on time, difficult times. If possible, feel free to create a PR to add an example, It will be helpful for others too.

Thanks.

**Reply by fahimbench** [reply]
Yeah i could when i ll have a few time to do it. I saw it was a recurrent question, i just followed some answers like to use schemas and stuffs like that. The only thing that i was stucked it was about the resize, you did it for pictures only, and we can't reuse it for another elements (useful for videos or iframe). I directly applied the same way, create a view above the element and a resizeView, its a few redundant.

**Reply by jorgemddev** [reply]
Hello, you know I'm having the same problem I want custom buttons, with DIV labels and design, but I don't know how to do it, can you explain how you did it for angular, please.

**Reply by jotacemarin** [reply]
Hi @fahimbench, can you share the code example in this thread please? 🙏

**Comment by rgustavsson** [comment]
Come on.. Delete this post or help us out :)

**Reply by sibiraj-s** [reply]
You can look at the example here https://github.com/sibiraj-s/ngx-editor/discussions/395 to create custom menu. WRT schema for video and audio, you should be able to refer prosemirror docs to create one. 

You can find examples of several schemas here too https://github.com/sibiraj-s/ngx-editor/tree/master/projects/ngx-editor/schema for reference.

**Reply by rgustavsson** [reply]
Thank you @sibiraj-s 
Do you have any samples of when external attributes need to be passed to the spec?

I tried adding an attrs object to:
`return setBlockType(schema.nodes.code_mirror, attrs)(state, dispatch);`

but these were not available in the spec toDom node object.

**Reply by rgustavsson** [reply]
Ok I got it working. I had not defined the neccessary attributes on the NodeSpec, and so once I defined them there I got it to work.

Cheers

**Reply by sibiraj-s** [reply]
great.

**Reply by sibiraj-s** [reply]
This entire library is build using features like these, in future, you should be able to browse around to find examples for most common requirements.

[a comment by an automated account is not shown]
